src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ConstructedType.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.Runtime;
+using Internal.TypeSystem;
+using Internal.IL;
+using Internal.Text;
+
+using ILCompiler.DependencyAnalysisFramework;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class ConstructedType : DependencyNodeCore<NodeFactory>, IEETypeNode
+    {
+        TypeDesc _type;
+
+        public ConstructedType(NodeFactory factory, TypeDesc type)
+        {
+            _type = type;
+            Debug.Assert(!type.IsCanonicalDefinitionType(CanonicalFormKind.Any));
+            CheckCanGenerateConstructedEEType(factory, type);
+        }
+
+        void ISymbolNode.AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append("ConstructedType: ");
+            sb.Append(_type.ToString());
+        }
+
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler) + " constructed";
+
+        public int Offset => 0;
+
+        public bool RepresentsIndirectionCell => false;
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        public TypeDesc Type => _type;
+
+        public override bool InterestingForDynamicDependencyAnalysis
```

Comment:
Generic virtual method analysis requires whole program view. How does copying this code from CoreRT type node help with dynamic statics?